iModelCore/iModelPlatform/DgnCore/CodeSpec.cpp

```diff
@@ -566,7 +568,26 @@ DgnCode DgnCode::FromJson(BeJsConst value, DgnDbCR db) {
         auto specPtr = db.CodeSpecs().GetCodeSpec(specJson.asCString());
         val.m_specId = specPtr.IsValid() ? specPtr->GetCodeSpecId() : CodeSpec::GetNullCodeSpecId();
     }
-    val.m_scope = value[json_scope()].asString();
-    val.m_value = value[json_value()].asString();
+
+    // the code scope must refer to an existing element. The json may have an ElementId or FederationGuid
+    auto scopeJson = value[json_scope()];
+    auto scopeStr = scopeJson.asCString();
+
+    // Note: this is necessary since GetId64 parses strings that start with number successfully and can get confused with Guids.
+    // ElementIds in JSON must start with "0x"
+    auto validId = scopeStr && (scopeStr[0] == '0' && (scopeStr[1] == 'X' || scopeStr[1] == 'x'));
+
+    auto scopeId = scopeJson.GetId64<DgnElementId>();
+    if (!validId || !db.Elements().ElementExists(scopeId)) {
+        BeGuid scopeFederationGuid;
+        DgnElementCPtr scopeEl;
+        if (BentleyStatus::SUCCESS == scopeFederationGuid.FromString(scopeJson.asCString()))
+            scopeEl = db.Elements().QueryElementByFederationGuid(scopeFederationGuid);

```

Comment:
You expressed some concerns about additional overhead imposed by verifying scope element exists. Are you ok with:
- Querying for and loading entire element vs just its federation Guid; and
- If caller supplies a valid but non-existent element Id, attempting fruitlessly to look up an element that has that Id as its federation Guid?

Reply:
> Querying for and loading entire element vs just its federation Guid;
This is rare and not worth worrying about.

>If caller supplies a valid but non-existent element Id, attempting fruitlessly to look up an element that has that Id as its federation Guid?
`scopeFederationGuid.FromString` will fail first.